### PR TITLE
Add example in .examples section

### LINKED ISSUE #63: Add example for preview generation dependencies
Some dependencies for efficient preview generation are missing. Preview is a pretty important feature for most users.

According to https://docs.nextcloud.com/server/11/admin_manual/installation/source_installation.html:
For preview generation (optional):
PHP module imagick
avconv or ffmpeg
OpenOffice or LibreOffice

### PR DESCRIPTION
**Pull request by Greek64**
This example adds dependencies for ALL optional nextcloud packages.
Also, the README.md was updated to detail the steps needed for each individual package, so users can easily modify the Dockerfile to their needs.

fixes #63

### CODE REVIEW

**Comment by Greek64** [comment]
I have no idea why the "gmp.h" is not found in the i386 variants, and I cannot test this locally.
Seems as if the i386 Systems need to explicitly install "libgmp3-dev:i386".
But how to know that from the Dockerfile?

File: .examples/README.md [hunk 1]
```diff
@@ -16,8 +16,51 @@ Example | Description
 [cron](https://github.com/nextcloud/docker/tree/master/.examples/dockerfiles/cron) | uses supervisor to run the cron job inside the container (so no extra container is needed).
 [imap](https://github.com/nextcloud/docker/tree/master/.examples/dockerfiles/imap) | adds dependencies required to authenticate users via imap
 [smb](https://github.com/nextcloud/docker/tree/master/.examples/dockerfiles/smb) | adds dependencies required to use smb shares
+[full](https://github.com/nextcloud/docker/tree/master/.examples/dockerfiles/full) | adds dependencies for ALL optional packages and cron functionality via supervisor (as in the `cron` example Dockerfile).
 
+### full
+The `full` Dockerfile example adds dependencies for all optional packages suggested by nextcloud that may be needed for some features (e.g. Video Preview Generation), as stated in the [Administration Manual](https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html).
```
Comment (tilosp) [inline .examples/README.md]: We should also add note that you have to enable the preview generation in the [config](https://docs.nextcloud.com/server/12/admin_manual/configuration_server/config_sample_php_parameters.html#previews) via `enabledPreviewProviders`. And add a warning that this opens your instance to [possible attack vectors](https://docs.nextcloud.com/server/12/admin_manual/configuration_server/harden_server.html?highlight=enabledpreviewproviders#disable-preview-image-generation).

**Comment by tilosp** [comment]
@Greek64 I fixed the i386 problem.
You can test the i386 variant by using `FROM i386/nextcloud` instead of `FROM nextcloud`.

**Comment by Greek64** [comment]
Thanks for the fix.
Added the requested notes, with links to the provided section in the Docs.

File: .examples/README.md (hunk 1 above)
Comment (Greek64) [reply .examples/README.md]: Added the requested notes, with links to the provided section in the Docs.

**Comment by tilosp** [comment]
Thank you very much for your contribution!

**Comment by pierreozoux** [comment]
Please don't think about supervisord for the real image.
Thanks!

**Comment by Greek64** [comment]
@pierreozoux 
Why though? (Not that I'm for it, but I would like to know the reason)

**Comment by pierreozoux** [comment]
@Greek64 https://github.com/docker-library/official-images#init

This is how I do the cron with compose:
https://github.com/indiehosters/nextcloud/blob/master/docker-compose.yml#L39-L53
(I know it is a bit dirty, but you know :) )

**Pull request merged**